### PR TITLE
feat(LERN-57): added that email will be stored and checked as lowercase in the backend

### PR DESCRIPTION
**Pull request by A1eeee**

### CODE REVIEW

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserService.kt [hunk 1]
```diff
@@ -30,11 +30,13 @@ public class UserService(
     public fun getAllAdminUsers(): List<UserDto> = userDao.fetchAllAdminUsers().map(::mapToDto)
 
     public fun createAdminUser(createUserDto: CreateUserDto) {
-        if (!isEmailValid(createUserDto.email)) {
+        val emailToLowerCase = createUserDto.email.trim().lowercase()
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserService.kt:33]: the `To` in the variabel name makes it sound like a function name. 
I would call it `emailLowerCase` or something similar.

File: backend/src/test/kotlin/ch/learntrack/backend/backoffice/user/UserIntegrationTest.kt [hunk 2]
```diff
@@ -165,4 +165,20 @@ class UserIntegrationTest: IntegrationTest() {
...
+        webClient.post()
+                .uri("/backoffice/user/create")
+                .body(BodyInserters.fromValue(validUser))
+                .setBasicAuthHeader(backendProperties)
+                .exchange()
+                .expectStatus()
+                .isOk
+
+        val user = userDao.fetchOne(USER.E_MAIL, validUser.email.lowercase())
+        assertThat(user).isNotNull
```
Comment (deepspace1000) [inline backend/src/test/kotlin/ch/learntrack/backend/backoffice/user/UserIntegrationTest.kt:181]: could you add an assert where you check the eMail?
`assertThat(user.email).isEqualTo(testlowercase@gmail.com)

**Review by deepspace1000 — CHANGES_REQUESTED** [review]
nice work

**Comment by A1eeee** [comment]
> nice work

Thank you. 
I am going to implement those changes when I am at home.

File: backend/src/test/kotlin/ch/learntrack/backend/backoffice/user/UserIntegrationTest.kt [hunk 3]
```diff
@@ -165,4 +165,21 @@ class UserIntegrationTest: IntegrationTest() {
...
+        assertThat(user?.eMail).isEqualTo(validUser.email.lowercase())
+    }
```
Comment (deepspace1000) [inline backend/src/test/kotlin/ch/learntrack/backend/backoffice/user/UserIntegrationTest.kt:183]: I personally would prefer to use a string here to be absolutely sure its working.

File: backend/src/test/kotlin/ch/learntrack/backend/backoffice/user/UserIntegrationTest.kt [hunk 4]
```diff
@@ -165,4 +165,21 @@ class UserIntegrationTest: IntegrationTest() {
                 .isEqualTo(HttpStatus.CONFLICT)
     }
 
+    @Test
+    fun `should be able to create a user even if email is in upper case`() {
```
Comment (deepspace1000) [inline backend/src/test/kotlin/ch/learntrack/backend/backoffice/user/UserIntegrationTest.kt:169]:
```suggestion
    fun `should save uppercase email as lowercase on user creation`() {
```
or something like that as it describes more what you are testing here.

File: backend/src/test/kotlin/ch/learntrack/backend/user/UserIntegrationTest.kt [hunk 5]
```diff
@@ -88,4 +88,21 @@ class UserIntegrationTest: IntegrationTest() {
             .expectStatus()
             .isUnauthorized
     }
+
+    @Test
+    fun `should return jws token even with uppercase email`() {
```
Comment (deepspace1000) [inline backend/src/test/kotlin/ch/learntrack/backend/user/UserIntegrationTest.kt:93]:
```suggestion
    fun `should return jws token with uppercase login email`() {
```

**Review by deepspace1000 — COMMENTED** [review]
just a bit more nitpicking 😉

File: backend/src/test/kotlin/ch/learntrack/backend/backoffice/user/UserIntegrationTest.kt (hunk 3 above)
Comment (deepspace1000) [reply backend/src/test/kotlin/ch/learntrack/backend/backoffice/user/UserIntegrationTest.kt:183]: thx 😄

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserService.kt [hunk 6]
```diff
@@ -30,11 +30,13 @@ public class UserService(
...
-        if (!isEmailValid(createUserDto.email)) {
+        val emailLowerCase = createUserDto.email.trim().lowercase()
+
+        if (!isEmailValid(emailLowerCase)) {
             throw LearnTrackConflictException("Invalid email")
         }
 
-        if (isEmailExisting(createUserDto.email)) {
+        if (isEmailExisting(emailLowerCase)) {
             throw LearnTrackConflictException("Email already exists")
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserService.kt:40]: i just saw this yet 😂. could you print the email in the Exception so if one occures  we understand why. you can use a template string like this.
```suggestion
            throw LearnTrackConflictException("Email ${emailLowerCase} already exists")
```

Comment (deepspace1000) [reply backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserService.kt:40]: also for the exception above too, the one with the invalid email

**Review by deepspace1000 — APPROVED** [review]
approved just implement the changes i requested, squash and merge.

**Pull request merged**